### DISCUSSION TITLE
Need to add MCU boot support in ESP32S3 with custom configuration.

### DISCUSSION THREAD

**Discussion by ankit1chauhan** [zephyrproject-rtos/zephyr#71420] [Q&A]
Hi,

I am using custom hardware running on ESP32S3 and an application that is currently working. I am trying to add the support of USB DFU on the existing ACM port which I am using as a shell and console.

I have followed the guide for using **sysbuild** that automatically adds the MCUBoot to the existing system. I confirm from the building logs that has below regions

   *****************************
   * Running CMake for mcuboot *
   *****************************

and 

   *************************
   * Running CMake for app *
   *************************
that successfully built the app and the **zephyr.bin** can be found in build/mcuboot/zephyr.

Now my question is I need to create a signed binary with a specific  .pem file encrypted on ED25519? Currently, I have the  .pem file generated using the ./imgtool.py keygen.

I have checked the configuration file in **bootloader. conf** which has different macros to use ED25519 encryption and to Use Serial through USB JTAG Serial port for Serial Recovery. I have tried to enable those macros but it did not affect anything in the image. 

Also, for DFU when I tried to add the mcumgr configuration in my prj.conf it shows the dependencies of ZEPHYR_ZCBOR_MODULE which is not configurable by the User.

So due to these two issues, I am not able to perform the USB DFU over the mcumgr utility. 

Below is my prj.conf

#Flash configurations
CONFIG_FLASH=y
CONFIG_FLASH_MAP=y
CONFIG_FLASH_PAGE_LAYOUT=y

#For mcumgr
CONFIG_MCUMGR=y
CONFIG_MCUMGR_TRANSPORT_UART=y
CONFIG_MCUMGR_TRANSPORT_SHELL=n
CONFIG_IMG_MANAGER=y

#Dependencies
CONFIG_NET_BUF=y
CONFIG_STREAM_FLASH=y
CONFIG_ZCBOR=y
CONFIG_BASE64=y
CONFIG_CRC=y
CONFIG_STATS=y
CONFIG_MCUMGR_GRP_IMG=y
CONFIG_MCUMGR_GRP_OS=y
CONFIG_MCUMGR_GRP_STAT=y
CONFIG_CBOR_API=y

**Comment by rftafas** [comment]
Hi!

FWIW, this is the first case I read about enabling encryption using Zephyr port on Espressif devices. Most people use Espressif Port and Secureboot V2. Can you confirm that it is not your case and you DO want MCUboot_Zephyr_Port Signature?

**Comment by ankit1chauhan** [comment]
Hi,

Thanks for reply.

Actually, I am looking for USB DFU support in ESP32 with Zephyr. I am currently using USB DFU using mcu manager and mcuboot with Nordic nRF. The hardware that I have has both NRF and ESP. So I was looking for a similar solution for ESP32 as well.

I am not familiar with the USB DFU procedure over ESP32 so I was grabbing the path that I have done for Nordic as Zephyr has a common platform for all vendors.

If there is a way I can have the same as you suggested then I can go with it. If you can share some references or an example for USB DFU on ESP32 using Zephyr that would be great.

**Reply by rftafas** [reply]
Let me add some comments:

- ESP32 has no USB port. Everything on its USB will eventually pass through some USB-Serial converter connected to UART0.
- USB for flashing without external device only exists on ESP32-C3 and newer.
- I can't say anything about using NordicSDK to build for Espressif. Don't know even if that doesn't fail any form of licensing.
- If this ESP32 was flashed with MCUboot, it should work with mcumgr. AFAIK, both Zephyr and Espressif port work.
- If you have an ESP32 device connected to another CPU (either Espressif or other), you can use [ESP Serial Flasher](https://github.com/espressif/esp-serial-flasher).

Please, take a look into the different ports of MCUboot also.

**Reply by ankit1chauhan** [reply]
Thanks for this information. I will check from my end what I can do.
